docs/preview/02-Features/writing-different-telemetry-types.md

```diff
@@ -429,6 +429,38 @@ logger.LogMetric("Invoice Received", 133.37, telemetryContext);
 
 ## Requests
 
+### Incoming Azure Service Bus requests
+Requests allow you to keep track of Azure Service Bus messages on a queue or topic and how the message was processed.
+
+Here is how you can report an Azure Service Bus queue request on a message that's being processed:
+
+```csharp
+using Microsoft.Extensions.Logging;
+
+bool isSuccessful = false;
+
+// Start measuring.
+using (var measurement = DependencyMeasurement.Start())
+{
+    try
+    {
+        // Processing message.
+
+        // End processing.
+        
+        isSuccessful = true;
+    }
+    finally
+    {
+        logger.LogServiceBusQueueRequest("<my-queue-namespace>.servicebus.windows.net", "<my-queue-name>", "<operation-name>", isSuccessful, measurement);
```

Comment:
I think @tomkerkhove  meant to pass in the namespace name of the ServiceBus, so without the `.servicebus.windows.net` part.

Actually, that would mean that the method signature would look like this:

```csharp
LogServiceBusQueueRequest( this ILogger logger,
    string namespacename,
    string queuename,
    string operationname,
    bool isSuccessful,
    DependencyMeasurement m,
    string namespaceSuffix = ".servicebus.windows.net
```
By doing that, you do not force developers to always add the '.servicebus.windows.net' part to their logs.  Moreover, I think they'll typically get the namespace-name from their Configuration (which probably doesn't contain that suffix).
If they must change the suffix, they can do it via the optional parameter.

Reply:
Best not to use optional parameters in public signatures of a library, as it's hard to maintain. We could create an overload for this, though.

Reply:
Added methods with `...WithPrefix` to make this distinction more clear.  https://github.com/arcus-azure/arcus.observability/pull/250/commits/9bacc6bd8c21cfddaa2decde9a9360d61a156254

Reply:
It's a suffix, and not a prefix :)

Reply:
Wel... you pass along the prefix, right, as in `{prefix}.servicebus.windows.net`. I remember a similar method overload in messaging where we do the same.

Reply:
The 'prefix' is what is mandatory to pass; the suffix (.servicebus.windows.net) is optional, as that may vary between Azure Clouds.
For the global Azure Cloud, this is `.servicebus.windows.net`.    In the German Azure Cloud, this is `servicebus.cloudapi.de`, for the Azure cloud in China this is yet another address (`servicebus.chinacloudapi.cn`).
So , that suffix is the part that should be optional.

Reply:
Yes, I got that, that's how we did it in messaging too. Hmm, I think we misunderstand each other. 😅 
Let's take a step back:

We have two type of methods, now. One that takes in the fully qualified name and one that only takes in the prefix.
These methods are called:
- `LogServiceBusRequest...()` 
- `LogServiceBusRequestWithPrefix...()`

This way we can take care of different clouds as well as giving consumers the possibility to manipulate the fully qualified name.

Reply:
Now I see; I just think that having this method named `...WithPrefix` adds to the confusion.  I'd just create it as a regular overload

Reply:
The problem is that we'll have signature conflicts, then, if we use the same method name. I used the same way of working as with Arcus Messaging, were we also include `...WithPrefix` in the method name when we want to provide a prefix instead of the full namespace. I don't see immediately a clearer solution besides renaming the method (like now). Moving the parameters around to get a different signature is probably not a very good developer experience and would maybe cause problems. With a different method name, we clearly state that we will use it differently.
Any suggestions on a different name, then? I wanted to be consistent but if it's causing confusion, then we could look for something else.

Reply:
It seems like there was a difference in interpretation; we took this offline.